pkg/metricscollector/v1alpha3/common/pns.go

```diff
@@ -95,6 +99,16 @@ func WaitPIDS(pids []int, opts ...WaitPidsOpts) error {
 				_, err := os.Stat(path)
 				if err != nil {
 					if os.IsNotExist(err) {
+						if opts[0].CompletedMarkedDirPath != "" {
```

Comment:
Cannot figure out how this helps solve the problem, could you please explain more about it?

Reply:
This PR adds `echo completed > $mountPath/$$$$.pid` after training command as below, if the training container succeed, it will touch a file named `$mountPath/$processID.pid` with "completed"
MetricsCollector container watches the training process, once find the process exit, it will check file `$mountPath/$processID.pid` to judge if the training process succeeds or not; if succeed, parse the metrics file; otherwise raise exception to exit.
Before this PR, metrics collector once find training process exit, it starts to parse metrics file, ignoring if training process exit status.

I tries to use "strace" to implement it, but it need more linux capability.
```
  - args:
    - python /mxnet/example/image-classification/train_mnist.py --batch-size=64 --lr=0.02273874688380991
      --num-layers=3 --optimizer=sgd 1>/var/log/katib/metrics.log 2>&1 && echo completed
      > /var/log/katib/$$$$.pid
    command:
    - sh
    - -c
```

Reply:
Will this cause the similar pipe exit code problem as tee does?

Reply:
no
1. if training process fails, it will return training process exit code ( `&& echo` will not be executed)
2. if training process succeeds (exit code is 0), `&& echo` will return 0, too. so the container exit code is 0, too

Reply:
Oh yeah, misunderstand the logic here, SGTM